### PR TITLE
[chore] Fix failing test case for flinkmetrics

### PR DESCRIPTION
**Pull request by MovieStoreGuy**
**Description:** 

On failure to read job metrics, this type can be nil causing a panic.

**Link to tracking Issue:**  NA

**Testing:**
NA

**Documentation:**
NA

### CODE REVIEW

**Comment by MovieStoreGuy** [comment]
I have yet to add tests to help get the release out sooner; happy to follow up in a future PR or leave it to code owners in an issue.

**Review by dmitryax — APPROVED** [review]

File: receiver/flinkmetricsreceiver/process.go
Comment (codeboten) [inline receiver/flinkmetricsreceiver/process.go:16]: Any idea what caused this to start failing now?

Comment (MovieStoreGuy) [reply receiver/flinkmetricsreceiver/process.go:16]: I suspect there was a change that was introduced but hasn't been guarded against in the lifecycle tests.

**Review by codeboten — APPROVED** [review]

**Pull request merged**
